pkg/build/node/node.go

```diff
@@ -415,6 +415,29 @@ func (c *BuildContext) prePullImages(dir, containerID string) error {
 		log.Errorf("Image build Failed! %v", err)
 		return err
 	}
+
+	// systemd-in-a-container should have read only /sys
+	// https://www.freedesktop.org/wiki/Software/systemd/ContainerInterface/
+	// however, we need other things from `docker run --privileged` ...
+	// and this flag also happens to make /sys rw, amongst other things
+	// Need --privileged flag in docker exec
+	if err = execInBuild("mount", "-o", "remount,ro", "/sys"); err != nil {
+		log.Warningf("Failed to mount /sys: %v", err)
+		return err
+	}
+	// kubernetes needs shared mount propagation
+	// Need --privileged flag in docker exec
+	if err = execInBuild("mount", "--make-shared", "/"); err != nil {
```

Comment:
again, this needs to happen on the actual node containers for mount propogation, which is something kubernetes uses. this can't be at build time unless docker added some new magic I don't know about :confused: 

same for the other mount below